Cabal/src/Distribution/Simple/Program/Ld.hs

```diff
@@ -67,6 +72,11 @@ combineObjectFiles verbosity lbi ld target files = do
   -- have a slight problem. What we have to do is link files in batches into
   -- a temp object file and then include that one in the next batch.
 
+  case Map.lookup "Supports relocatable output" $ programProperties ld of
+    Just "YES" -> pure ()
+    Just "NO" ->
+      dieWithException verbosity $ CombineObjectFilesRelocatableUnsupported
+    _other -> pure () -- This may still fail if the compiler does not support -r.
```

Comment:
Sorry for barging in at the eleventh hour, this is not a blocking request but would you mind annotating this comment with `NOTE`?